autoscale/src/main/java/com/sequenceiq/periscope/monitor/TimeMonitor.java

```diff
@@ -25,4 +28,10 @@ public String getTriggerExpression() {
         return CronTimeEvaluator.class;
     }
 
+    @Override
+    protected List<Cluster> getMonitored() {
+        List<Long> clusterIds = getClusterService().findTimeAlertClustersForNode(StackType.WORKLOAD,
```

Comment:
is it better in performance if you query the ids first and query the clusters by the ids second?

Reply:
yes because of the join with timealert table. also only cluster ids which have schedule based alerts are filtered in monitor instead of loading all clusters with autoscaling enabled and processing them in CronTimeEvaluator